scipy/stats/_bws_test.py

```diff
@@ -0,0 +1,230 @@
+        pvalue : float
+            The associated *p*-value for the chosen `alternative`.
+    See Also
+    --------
+    scipy.stats.wilcoxon, scipy.stats.mannwhitneyu, scipy.stats.ttest_ind
+
+    References
+    ----------
+    .. [1] Neuhäuser, M. (2005). Exact Tests Based on the Baumgartner-Wei-Schindler 
+           StatisticA Survey. Statistical Papers, 46(1), 1-29.
+    .. [2] Shan, G., Ma, C., Hutson, A. D., & Wilding, G. E. (2013). Some tests 
+           for detecting trends based on the modified Baumgartner–Weiß–Schindler 
+           statistics. Computational Statistics & Data Analysis, 57(1), 246-261.
+    .. [3] Fay, M. P., & Proschan, M. A. (2010). Wilcoxon-Mann-Whitney or t-test? 
+           On assumptions for hypothesis tests and multiple interpretations of 
+           decision rules. Statistics surveys, 4, 1.
+    .. [4] Murakami, H. (2006). A k-sample rank test based on modified Baumgartner
+           statistic and its power comparison. Journal of the Japanese Society of 
+           Computational Statistics, 19(1), 1-13.
+    .. [5] Neuhäuser, M. (2001). One-sided two-sample and trend tests based on a 
+           modified Baumgartner-Weiß-Schindler statistic. Journal of Nonparametric 
+           Statistics, 13(5), 729-739.
+    .. [6] Good, P. (2013). Permutation tests: a practical guide to resampling 
+           methods for testing hypotheses. Springer Science & Business Media.
+
+    Examples
+    --------
+    We follow the example of table 3 in [1]_: Fourteen children were divided randomly into two groups, 
+    an experimental group and a control group. For each child the number of correct identifications 
+    among 18 new examples was observed.
+
+    >>> control_group_ranks = np.array([1, 2, 3, 4, 6, 7, 8])
+    >>> experimental_group_ranks = np.array([5, 9, 10, 11, 12, 13, 14])
+
+    We use the BWS test to assess whether there is a 
+    statistically significant difference in the methods of training preschool children.
+    The null hypothesis is that the distribution of first method is the same as 
+    the distribution of second one. We decide that a confidence level of 95% is 
+    required to reject the null hypothesis in favor of the alternative that 
+    the distributions are different.
+    Since the number of samples is very small and there are no ties in the
+    data, we can compare the observed test statistic against the *exact*
+    distribution of the test statistic under the null hypothesis.
+
+    >>> from scipy.stats import bws_test
+    >>> B, p = bws_test(control_group_ranks, experimental_group_ranks)
+    >>> print(B)
+    5.132167152575315
+
+    This agrees with :math:`B = 5.132` reported in [1]_. The *p*-value produced
+    by `bws_test` agrees with :math:`p = 0.0029` reported in [1]_.
+
+    >>> print(p)
+    0.002913752913752914
+
+    It normally takes a long time to calculate the exact *p*-value when the passing arrays
```

Comment:
The default number if resamples is 9999. If an exact test can be performed with fewer samples, it will be. Otherwise, a randomized test is performed.